lua/telekasten.lua

```diff
@@ -1442,38 +1442,48 @@ local function InsertLink(opts)
         return
     end
 
-    find_files_sorted({
-        prompt_title = "Insert link to note",
-        cwd = M.Cfg.home,
-        attach_mappings = function(prompt_bufnr, map)
-            actions.select_default:replace(function()
-                actions.close(prompt_bufnr)
-                local selection = action_state.get_selected_entry()
-                local pinfo = Pinfo:new({
-                    filepath = selection.value,
-                    opts,
-                })
-                vim.api.nvim_put(
-                    { "[[" .. pinfo.title .. "]]" },
-                    "",
-                    true,
-                    true
-                )
-                if opts.i then
-                    vim.api.nvim_feedkeys("A", "m", false)
-                end
-            end)
-            map("i", "<c-y>", picker_actions.yank_link(opts))
-            map("i", "<c-i>", picker_actions.paste_link(opts))
-            map("n", "<c-y>", picker_actions.yank_link(opts))
-            map("n", "<c-i>", picker_actions.paste_link(opts))
-            map("i", "<c-cr>", picker_actions.paste_link(opts))
-            map("n", "<c-cr>", picker_actions.paste_link(opts))
-            return true
-        end,
-        find_command = M.Cfg.find_command,
-        sort = M.Cfg.sort,
-    })
+    local cwd = M.Cfg.home
+    local find_command = M.Cfg.find_command
+    local sort = M.Cfg.sort
+    local attach_mappings = function(prompt_bufnr, map)
+        actions.select_default:replace(function()
+            actions.close(prompt_bufnr)
+            local selection = action_state.get_selected_entry()
+            local pinfo = Pinfo:new({
+                filepath = selection.filename,
+                opts,
+            })
+            vim.api.nvim_put({ "[[" .. pinfo.title .. "]]" }, "", true, true)
+            if opts.i then
+                vim.api.nvim_feedkeys("A", "m", false)
+            end
+        end)
+        map("i", "<c-y>", picker_actions.yank_link(opts))
+        map("i", "<c-i>", picker_actions.paste_link(opts))
+        map("n", "<c-y>", picker_actions.yank_link(opts))
+        map("n", "<c-i>", picker_actions.paste_link(opts))
+        map("i", "<c-cr>", picker_actions.paste_link(opts))
+        map("n", "<c-cr>", picker_actions.paste_link(opts))
+        return true
+    end
+
+    if opts.with_live_grep then
```

Comment:
I worry about option field value. Current implementation with `with_live_grep` option is Boolean field. But, if we will add some other pickers, we should use String field. As show below

 ```suggestion
    if opts.picker == 'buildin_live_grep' then
```

Reply:
This works for now, so I am ok with what is in the PR. 

I think it would be a good idea to massively refactor Telekasten at some point. We would then optimize the code of the pickers and streamline the options. But in the meantime, I am ok with using a boolean field...